agent/src/softwarecontaineragent.cpp

```diff
@@ -223,8 +173,9 @@ ContainerID SoftwareContainerAgent::createContainer(const std::string &config)
     profilefunction("createContainerFunction");
 
     try {
-        parseConfig(config);
-    } catch (SoftwareContainerError &err) {
+        ContainerConfigParser::ContainerOptions options = m_containerParser.parseConfig(config);
+        m_containerConfig.setEnableWriteBuffer(options.enableWriteBuffer);
+    } catch (ContainerConfigParseError &err) {
```

Comment:
Since ContainerConfigParseError is internal to SC and should inherit SoftwareContainerError, and since we catch the outermost base class, I think we could skip catching and throwing here. If we inherit the base class and initialize it with the message already when creating the original exception, we would maintain the same log abilities.